stdlib/thread_local_storage.mli

```diff
@@ -0,0 +1,72 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*      KC Sivaramakrishnan, Indian Institute of Technology, Madras       *)
+(*                 Stephen Dolan, University of Cambridge                 *)
+(*                   Tom Kelly, OCaml Labs Consultancy                    *)
+(*                                                                        *)
+(*   Copyright 2019 Indian Institute of Technology, Madras                *)
+(*   Copyright 2014 University of Cambridge                               *)
+(*   Copyright 2021 OCaml Labs Consultancy Ltd                            *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Thread-local Storage.
+
+    @since 5.2 *)
+
+module Key : sig
```

Comment:
Why do you need a `Key` submodule ? This is a reference cell. The following seems entirely sufficient. 

```ocaml
module Thread_local_storage : sig 
  type 'a t
  (** The type for thread local storage for a value of type ['a]. *)

  val make : ?split_from_parent:('a -> 'a) -> (unit -> 'a) -> 'a t
  val get : 'a t -> 'a 
  val set : 'a t -> 'a -> unit
end
```

Also it's `make` not `create` since the data is initialized (very obscure convention but that's what it is, see the discussion in https://github.com/ocaml/ocaml/pull/9921).